feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go

```diff
@@ -537,18 +580,281 @@ func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good, bad, lb []
+	t.Logf("Adding %d ipv4 vrf prefixes and gribi entries for them", vrfPrefixcount)
+
+	prefixes = []string{}
+	for s := 0; s < vrfPrefixcount; s++ {
+		prefixes = append(prefixes, generateIPAddress(dstPfxvrf1, s))
+	}
+	for _, prefix := range prefixes {
+		ipv4Entry := fluent.IPv4Entry().
+			WithNetworkInstance(vrfName).
+			WithPrefix(prefix + "/" + mask).
+			WithNextHopGroup(uint64(nhg20ID)).
+			WithNextHopGroupNetworkInstance((deviations.DefaultNetworkInstance(args.dut)))
+		args.client.Fluent(t).Modify().AddEntry(t, ipv4Entry)
+	}
+	if err := args.client.AwaitTimeout(context.Background(), t, 2*time.Minute); err != nil {
+		t.Fatalf("Error waiting to add IPv4: %v", err)
+	}
+
+	grv, err := args.client.Fluent(t).Get().
+		WithNetworkInstance(vrfName).
+		WithAFT(fluent.IPv4).
+		Send()
+
+	if err != nil {
+		t.Fatalf("got unexpected error from get, got: %v", err)
+	}
+
+	for _, prefix := range prefixes {
+		chk.GetResponseHasEntries(t, grv,
+			fluent.IPv4Entry().
+				WithNetworkInstance(vrfName).
+				WithNextHopGroup(uint64(nhg1ID)).
+				WithPrefix(prefix+"/"+mask),
+		)
+	}
+
+	t.Log("Create flows for port1 to port2, port1 to port3 and port1 to port4")
+
+	p2flow := createFlow("Flow Port 1 to Port 2", args.ate, args.top, true, &atePort2)
+	p3flow := createFlow("Flow Port 1 to Port 3", args.ate, args.top, true, &atePort3)
+	p4flow := createFlow("Flow Port 1 to Port 4", args.ate, args.top, true, &atePort4)
+
+	t.Log("Validate primary path traffic recieved at ate port2, ate port3 and no traffic on ate port4")
+	validateTrafficFlows(t, args.ate, nil, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, 1, args.client, false)
+
+	t.Logf("Adding NH %d for trunk4 via gribi", nh3ID)
+	if deviations.GRIBIMACOverrideWithStaticARP(args.dut) {
+		args.client.AddNH(t, nh3ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk4, Mac: port4mac, Dest: atePort4.IPv4})
+	} else {
+		args.client.AddNH(t, nh3ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk4, Mac: port4mac})
+	}
+
+	t.Log("Validate traffic switching from  ate port2, ate port3 to ate port4")
+	validateTrafficFlows(t, args.ate, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, nil, 2, args.client, true)
+
+	t.Log("Validate traffic switching from  ate port4 back to ate port2 and ate port3")
+	validateTrafficFlows(t, args.ate, nil, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, 3, args.client, true)
+
+}
+
+// deleteinterfaceconfig unconfigs interfaces.
+func deleteinterfaceconfig(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	p2 := dut.Port(t, "port2")
+	p3 := dut.Port(t, "port3")
+	p4 := dut.Port(t, "port4")
+
+	gnmi.Delete(t, dut, d.Interface(p2.Name()).Config())
```

Comment:
can we delete only subinterface as port config is needed for lag configuration .

	gnmi.Delete(t, dut, d.Interface(p2.Name()).SubInterface(0).Config())
	gnmi.Delete(t, dut, d.Interface(p3.Name()).SubInterface(0).Config())
	gnmi.Delete(t, dut, d.Interface(p4.Name()).SubInterface(0).Config())

Reply:
Thought to have a cleaner way of deleting ports configs before associating to bundle, and so had it. Have changed the deletes to subinterface now.